code/media/src/main/java/com/adobe/marketing/mobile/media/internal/MediaDatabase.java

```diff
@@ -43,8 +43,14 @@ class MediaDatabase {
         if (context == null) {
             throw new IllegalStateException("Context is null");
         }
-        File database = context.getDatabasePath(dbName);
-        dbPath = database.getPath();
+        dbPath = context.getDatabasePath(dbName);
+
+        File databaseDir = dbPath.getParentFile();
+        // Create the parent directory if it does not exist
+        if (databaseDir != null && !databaseDir.exists()) {
+            databaseDir.mkdirs();
```

Comment:
https://developer.android.com/reference/java/io/File?hl=en#mkdirs()
`mkdirs()` may throw `SecurityException`, can we add a try/catch here?

Reply:
The parent class handles all the exceptions.